pymc3/sampling.py

```diff
@@ -453,41 +596,52 @@ def init_nuts(init='ADVI', njobs=1, n_init=500000, model=None,
         Instantiated and initialized NUTS sampler object
     """
 
-    model = pm.modelcontext(model)
-
-    pm._log.info('Initializing NUTS using {}...'.format(init))
-
-    random_seed = int(np.atleast_1d(random_seed)[0])
+    start, cov = do_init(init, njobs, n_init, model, random_seed)
+    step = pm.NUTS(scaling=cov, is_cov=True, **kwargs)
+    return start, step
 
-    if init is not None:
-        init = init.lower()
 
-    if init == 'advi':
-        v_params = pm.variational.advi(n=n_init, random_seed=random_seed)
-        start = pm.variational.sample_vp(v_params, njobs, progressbar=False,
-                                         hide_transformed=False,
-                                         random_seed=random_seed)
-        if njobs == 1:
-            start = start[0]
-        cov = np.power(model.dict_to_array(v_params.stds), 2)
-    elif init == 'advi_map':
-        start = pm.find_MAP()
-        v_params = pm.variational.advi(n=n_init, start=start,
-                                       random_seed=random_seed)
-        cov = np.power(model.dict_to_array(v_params.stds), 2)
-    elif init == 'map':
-        start = pm.find_MAP()
-        cov = pm.find_hessian(point=start)
-    elif init == 'nuts':
-        init_trace = pm.sample(step=pm.NUTS(), draws=n_init,
-                               random_seed=random_seed)[n_init // 2:]
-        cov = np.atleast_1d(pm.trace_cov(init_trace))
-        start = np.random.choice(init_trace, njobs)
-        if njobs == 1:
-            start = start[0]
-    else:
-        raise NotImplemented('Initializer {} is not supported.'.format(init))
+def get_random_starters(nwalkers, model=None):
```

Comment:
Can this only be used with a particle sampler? Seems like the function is more general than that.